cli/src/cmd/cast/send.rs

```diff
@@ -81,14 +85,59 @@ impl SendTxArgs {
             json: to_json,
             resend,
             command,
+            unlocked,
         } = self;
         let config = Config::from(&eth);
         let provider = utils::get_provider(&config)?;
         let chain = utils::get_chain(config.chain_id, &provider).await?;
         let api_key = config.get_etherscan_api_key(Some(chain));
         let mut sig = sig.unwrap_or_default();
 
-        if let Ok(signer) = eth.wallet.signer(chain.id()).await {
+        let code = if let Some(SendTxSubcommands::Create {
+            code,
+            sig: constructor_sig,
+            args: constructor_args,
+        }) = command
+        {
+            sig = constructor_sig.unwrap_or_default();
+            args = constructor_args;
+            Some(code)
+        } else {
+            None
+        };
+
+        // Case 1:
+        // Default to sending via eth_sendTransaction if the --unlocked flag is passed.
+        // This should be the only way this RPC method is used as it requires a local node
+        // or remote RPC with unlocked accounts.
+        if unlocked {
+            // Checking if signer isn't the default value
+            // 00a329c0648769A73afAc7F9381E08FB43dBEA72.
+            if resend {
+                tx.nonce = Some(provider.get_transaction_count(config.sender, None).await?);
+            }
+
+            cast_send(
```

Comment:
`cast_send` is called twice. 
Since `cast_send` is our final intent, can we refactor to have just a call at the end of the fn block?

Reply:
Yeah the reason this is written like so is that `provider` actually takes different types depending if you have a signer or not, which will also affect `from` and the `eth_*` method used. We want to clearly differentiate when we're falling back to `eth_sendTransaction` vs using a signer and using `eth_sendRawTransaction`, and so we have these two code paths. I actually think abstracting it all away so it looks like only one path could make this harder to reason about and could allow misleading error messages due to fall-through cases. Let me know what you think!